parsley-debug/shared/src/main/scala/parsley/debug/Codec.scala

```diff
@@ -44,6 +44,20 @@ trait Codec[A] {
 }
 
 object Codec {
+
+    /** The implicit conversion of a reference to an encodable object
+      * 
+      * @constructor This constructor should not be called manually, it is designed to be used via Scala's implicit resolution.
+      * @param r The reference
+      * @param c The string encoder-decoder for type `B`
+      */
+    implicit class RefOps[B](r: Ref[B])(implicit c: Codec[B]) extends RefCodec {
```

Comment:
Traditionally, we'd use an `implicit def` returning a new anonymous `RefCodec` object. Implicit classes are used for extension methods, which this isn't.